pkg/local_object_storage/engine/engine.go

```diff
@@ -163,7 +164,9 @@ func (e *StorageEngine) reportShardErrorWithFlags(
 		zap.String("error", err.Error()),
 	}, fields...)...)
 
-	if e.errorsThreshold == 0 || errCount < e.errorsThreshold {
+	if errors.Is(err, common.ErrFatal) {
+		e.log.Error("fatal error encountered", zap.Error(err))
```

Comment:
maybe also ignore `block` in an exceptional case?

Reply:
No, we could dead-lock in that case.